packages/shared/friends/sagas.ts

```diff
@@ -388,22 +403,218 @@ function* refreshFriends() {
+}
+
+export async function getPrivateMessages(getPrivateMessagesPayload: GetPrivateMessagesPayload) {
+  const client: SocialAPI | null = getSocialClient(store.getState())
+  if (!client) return
+
+  // get the conversation.id
+  const conversationId = await getConversationId(client, getPrivateMessagesPayload.userId)
+
+  const ownId = client.getUserId()
+
+  // get cursor of the conversation located on the given message or at the end of the conversation if there is no given message.
+  const messageId: string | undefined = !getPrivateMessagesPayload.fromMessageId
+    ? undefined
+    : getPrivateMessagesPayload.fromMessageId
+
+  // the message in question is in the middle of a window, so we multiply by two the limit in order to get the required messages.
+  let limit = getPrivateMessagesPayload.limit
+  if (messageId !== undefined) {
+    limit = limit * 2
+  }
+
+  const cursorMessage = await client.getCursorOnMessage(conversationId, messageId, {
+    initialSize: limit,
+    limit
+  })
+
+  const messages = cursorMessage.getMessages()
+  if (messageId !== undefined) {
+    // we remove the messages they already have.
+    const index = messages.map((messages) => messages.id).indexOf(messageId)
+    messages.splice(index)
+  }
+
+  // parse messages
+  const addChatMessagesPayload: AddChatMessagesPayload = {
+    messages: messages.map((message) => ({
+      messageId: message.id,
+      messageType: ChatMessageType.PRIVATE,
+      timestamp: message.timestamp,
+      body: message.text,
+      sender: message.sender === ownId ? getUserIdFromMatrix(ownId) : getPrivateMessagesPayload.userId,
+      recipient: message.sender === ownId ? getPrivateMessagesPayload.userId : getUserIdFromMatrix(ownId)
+    }))
+  }
 
-    yield Promise.all(Object.values(socialInfo).map(({ userId }) => ensureFriendProfile(userId))).catch(logger.error)
+  getUnityInstance().AddChatMessages(addChatMessagesPayload)
+}
+
+export function getUnseenMessagesByUser() {
+  const conversationsWithMessages = getAllConversationsWithMessages(store.getState())
+
+  if (conversationsWithMessages.length === 0) {
+    return
+  }
 
-    const initMessage = {
-      currentFriends: friendIds,
-      requestedTo: requestedToIds,
-      requestedFrom: requestedFromIds
+  const updateTotalUnseenMessagesByUserPayload: UpdateTotalUnseenMessagesByUserPayload = { unseenPrivateMessages: [] }
+
+  for (const conversation of conversationsWithMessages) {
+    // unseenPrivateMessages[conversation.conversation.userIds![1]] = conversation.conversation.unreadMessages?.length || 0
+    updateTotalUnseenMessagesByUserPayload.unseenPrivateMessages.push({
+      count: conversation.conversation.unreadMessages?.length || 0,
+      userId: conversation.conversation.userIds![1]
+    })
+  }
+
+  getUnityInstance().UpdateTotalUnseenMessagesByUser(updateTotalUnseenMessagesByUserPayload)
+}
+
+export function getFriendsWithDirectMessages(request: GetFriendsWithDirectMessagesPayload) {
```

Comment:
- We have some console logs here, not sure if we want to keep them. 
- Also, I think we're not lazy loading the list of conversations. I mean we should be making use of request.limit and request.skip, shouldn't we?

Reply:
Let's go ahead and remove all `console#log` as well as any commented out code.